File: exercise-4/Monster.js
class Monster {
  constructor(name, nickname, teeth, demeanor, location) {
    this.name = name;
    this.nickname = nickname;
    this.teeth = teeth;
    this.demeanor = demeanor;
    this.location = location;
    this.isScary = true;
  }

  roar() {
    console.log('RARRRR')
  }

  changeName(newName) {
    this.name = newName
    return `I go by ${this.name} now!`
  }

  visitDentist(newTeeth) {
    this.teeth = newTeeth
  }

  changeMood(newMood) {
    this.demeanor = newMood
    console.log("Now all of a sudden I feel very " + this.demeanor + "!")
  }
}

File: exercise-4/Monster.rb
class Monster
  def initialize(name, nickname, teeth, demeanor, location)
    @name        = brand
    @nickname    = nickname
    @teeth       = teeth;
    @demeanor    = demeanor;
    @location    = location;
    @is_scary    = true;
  end

  def roar
    puts "RARRR"
  end

  def change_name(new_name)
    @name = new_name
    return "I got by #{@name} now!"
  end

  def visit_dentist(new_teeth)
    @teeth = new_teeth
  end

  def change_mood(new_mood)
    @demeanor = new_mood
    puts "Now all of a sudden I feel very #{@demeanor}!"
  end
end

